### PR TITLE
cdc: remove unnecessary goroutines

### PR DESCRIPTION
**Pull request by overvenus**
### What problem does this PR solve? 

Remove unnecessary goroutines by multiplexing the select statement. Also, this PR reuse timers for flushing metrics.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/pipeline/sorter.go [hunk 1]
```diff
@@ -110,6 +102,10 @@ func (n *sorterNode) Init(ctx pipeline.NodeContext) error {
 		for {
 			select {
 			case <-stdCtx.Done():
+				// Since the flowController is implemented by `Cond`, it is not
+				// cancelable by a context. We need to listen on cancellation
+				// and aborts the flowController manually.
+				n.flowController.Abort()
```
Comment (liuzix) [inline cdc/processor/pipeline/sorter.go:108]: Aborting the `flowController` here could cause deadlock

File: cdc/entry/mounter.go [hunk 2]
```diff
@@ -150,16 +147,18 @@ func (m *mounterImpl) collectMetrics(ctx context.Context) {
 	changefeedID := util.ChangefeedIDFromCtx(ctx)
 	metricMounterInputChanSize := mounterInputChanSizeGauge.WithLabelValues(captureAddr, changefeedID)
 
+	metricsTimer := time.NewTimer(defaultMetricInterval)
```
Comment (ben1009) [inline cdc/entry/mounter.go:150]: defer stop()

Comment (ben1009) [reply cdc/entry/mounter.go:150]: NewTimer or NewTicker ?

Comment (overvenus) [reply cdc/entry/mounter.go:150]: I prefer Timer because Ticker buffers timeout internally that may lead to record metrics twice in a very short period.

**Pull request closed without merging**